packages/api/src/rich-text/detection.ts

```diff
@@ -69,6 +69,33 @@ export function detectFacets(text: UnicodeString): Facet[] | undefined {
       })
     }
   }
+  {
+    const re = /(?:^|\s)(#[^\d\s]\S*)(?=\s)?/g
+    while ((match = re.exec(text.utf16))) {
+      let [tag] = match
+      const hasLeadingSpace = /^\s/.test(tag)
```

Comment:
Matches a leading space rn because I need to make sure that something like `https://esb.lol/page#hash` doesn't match, but `I love #turtles` does. So this is needed rn to bump the index start so that we ignore the leading space.

Lmk if y'all have regex ideas and I can provide some text to test with.

Reply:
Might be possible with a lookahead or lookbehind but not sure it's worth bothering with